eng/pipelines/templates/steps/build.yml

```diff
@@ -60,28 +60,25 @@ steps:
       Artifacts: ${{ parameters.Artifacts }}
       ArtifactPath: $(Build.SourcesDirectory)/sdk/${{ parameters.ServiceDirectory }}
 
-  # It's important for performance to pass "sdk" as "sourceFolder" rather than as a prefix in "contents".
-  # The task first enumerates all files under "sourceFolder", then matches them against the "contents" pattern.
-  - task: CopyFiles@2
-    inputs:
-      sourceFolder: sdk
-      contents: |
-        **/$(coalesceResultFilter)/*.tgz
-        **/$(coalesceResultFilter)/browser/*.zip
-      targetFolder: $(Build.ArtifactStagingDirectory)
-      flattenFolders: true
-    displayName: "Copy packages"
-
-  - template: /eng/common/pipelines/templates/steps/publish-artifact.yml
-    parameters:
-      ArtifactPath: '$(Build.ArtifactStagingDirectory)'
-      ArtifactName: 'packages'
-
   - template: ../steps/generate-doc.yml
     parameters:
       ServiceDirectory: ${{parameters.ServiceDirectory}}
 
+  - pwsh: |
+      $artifacts = '${{ convertToJson(parameters.Artifacts) }}' | ConvertFrom-Json
+      foreach ($artifact in $artifacts)
+      {
+        $artifactName = $artifact.name
+        Write-Host "Copying $artifactName artifacts to $(Build.ArtifactStagingDirectory)/$artifactName"
+        New-Item -Type Directory -Name $artifactName -Path $(Build.ArtifactStagingDirectory)
```

Comment:
nit: It might be worth piping these new directory calls to null to avoid the extra noise in the logs. see https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/scripts/Helpers/PSModule-Helpers.ps1#L42 as an example.